pylint/checkers/classes.py

```diff
@@ -930,11 +937,18 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
                     ):
                         break
             else:
+                name_stack = ""
+                curr = function_def.parent.scope()
+                # Generate proper names for nested functions
+                while curr != node:
+                    name_stack = f"{curr.name}." + name_stack
```

Comment:
I don't like the string concatenation here. It would be better to add each name to a list and then do something like `".".join(reversed(name_stack))`